host_provider/providers/azure.py

```diff
@@ -0,0 +1,342 @@
+from collections import namedtuple
+from requests.status_codes import codes as response_created
+import os
+import json
+import re
+from collections import OrderedDict
+from contextlib import suppress
+from host_provider.credentials.azure import CredentialAddAzure, CredentialAzure
+from host_provider.common.azure import AzureConnection
+from host_provider.models import Host
+from host_provider.providers import ProviderBase
+from host_provider.settings import HOST_ORIGIN_TAG
+from libcloud.compute.types import Provider
+from host_provider.clients.team import TeamClient
+from pathlib import Path
+
+
+class OperationNotAllowed(Exception):
+    pass
+
+
+class DeployVmError(Exception):
+    pass
+
+
+class OfferingNotFoundError(Exception):
+    pass
+
+
+class InvalidParameterError(Exception):
+    pass
+
+
+class JsonTemplates(object):
+    def __init__(self, path="host_provider/templates/azure/version/"):
+        self.path = path
+
+    def list_files(self, version):
+        files_version = os.path.join(self.path, f"{version}/")
+        files = [path for path in Path(files_version).rglob('*.json')]
+        return files
+
+    def load_json(self, json_file):
+        data = None
+
+        with open(json_file) as fp:
+            data = json.load(fp)
+        return data
+
+    def dumps_json(self, obj):
+        if isinstance(obj, dict):
+            return json.dumps(obj)
+        return None
+
+
+class AzureProvider(ProviderBase):
+    BasicInfo = namedtuple("AzureBasicInfo", "id")
+    azClient = None
+    connCls = AzureConnection
+
+    @classmethod
+    def get_provider(cls):
+        return Provider.AZURE_ARM
+
+    def build_client(self):
+        AzureClient = self.get_driver()
+
+        client = AzureClient(
+            self.credential.tenant_id,
+            self.credential.subscription_id,
+            self.credential.access_id,
+            self.credential.secret_key,
+            region=self.credential.region
+        )
+        return client
+
+    def build_credential(self):
+        return CredentialAzure(
+            self.get_provider(), self.environment, self.engine)
+
+    def get_azure_connection(self):
+        az = self.connCls()
+        self.azClient = az
+
+        return az.conn_cls(self.credential.access_id,
+                           self.credential.secret_key,
+                           tenant_id=self.credential.tenant_id,
+                           subscription_id=self.credential.subscription_id)
+
+    def get_node(self, node_id):
+        pass
+
+    def _parse_image(self, name, size, gallery='myGallery', image='mssql_2019_0_0', version='1.0.0'):
+        sql_dict = OrderedDict()
+        templates = JsonTemplates()
+        pw = self.credential.init_password
+        region = self.credential.region
+
+        image_id = '/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s' \
```

Comment:
- Hardcoded url.
- use format instead of %
- do no use \ to break lines, use parentesis